hr_attendance_management/models/hr_forced_due_hours.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2018 Compassion CH (http://www.compassion.ch)
+#    @author: Samuel Fringeli <samuel.fringeli@me.com>
+#
+#    The licence is in the file __manifest__.py
+#
+##############################################################################
+
+from odoo import models, fields, api
+
+
+class HrForcedDueHours(models.Model):
+    _name = 'hr.forced.due.hours'
+    _description = "HR forced due hours"
+    _order = 'date'
+
+    _sql_constraints = [('unique_due_hours', 'unique(date, employee_id)',
+                         'This "Forced due hour" already exists')]
+
+    employee_id = fields.Many2one('hr.employee',
+                                  string='Employee', required=True)
+    date = fields.Date('Date', required=True)
+    forced_due_hours = fields.Float('Forced due hours', required=True)
+
+    @api.model
+    def recompute_due_hours(self, employee_id, date):
+        return self.env['hr.attendance.day'].search(
+            [('employee_id', '=', employee_id),
+             ('date', '=', date)]).recompute_due_hours()
+
+    @api.multi
+    def write(self, vals):
+        for record in self:
```

Comment:
Ok... could you explain me the reasons?

Reply:
Performance will be very bad when writing on multiple records as best explained by Odoo. 
Have a look at https://www.youtube.com/watch?v=dQYGau81RYg (slides here : https://www.odoo.com/slides/slide/orm-performance-optimizations-best-practices-506).
Part two is what applies here because we're already in a O(n) for 'simple' writes and with the successive calls to `recompute_due_hours` we'll end up having lots of processing and I won't calculate here how many requests this would imply. :wink: